pkg/controllers/machine/liveness.go

```diff
@@ -16,30 +16,73 @@ package machine
 
 import (
 	"context"
+	"time"
 
+	"github.com/prometheus/client_golang/prometheus"
+	"go.uber.org/multierr"
 	"k8s.io/utils/clock"
+	"knative.dev/pkg/logging"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 	"sigs.k8s.io/controller-runtime/pkg/reconcile"
 
 	"github.com/aws/karpenter-core/pkg/apis/settings"
 	"github.com/aws/karpenter-core/pkg/apis/v1alpha5"
+	"github.com/aws/karpenter-core/pkg/metrics"
+	"github.com/aws/karpenter-core/pkg/utils/result"
 )
 
 type Liveness struct {
 	clock      clock.Clock
 	kubeClient client.Client
 }
 
+// creationTTL is a heuristic time that we expect to succeed with our cloudprovider.Create() call
+// If we don't succeed within this time, then we should delete and try again through some other mechanism
+const creationTTL = time.Minute * 2
+
 func (l *Liveness) Reconcile(ctx context.Context, machine *v1alpha5.Machine) (reconcile.Result, error) {
+	creationRes, creationErr := l.creationTTL(ctx, machine)
+	registrationRes, registrationErr := l.registrationTTL(ctx, machine)
+	return result.Min(creationRes, registrationRes), multierr.Combine(creationErr, registrationErr)
```

Comment:
I don't know if this is how it works, but can you add a `reconcile.Result{Requeue: true}` to the `result.Min` calls so that we get to take advantage of the controller runtime backoff, but if the backoff exceeds when the TTL will trigger, it'd return the TTL time?